src/applications/check-in/utils/navigation/pre-check-in/index.js

```diff
@@ -44,23 +44,66 @@ const PRE_CHECK_IN_FORM_PAGES = Object.freeze([
   },
 ]);
 
-const createForm = ({
-  hasConfirmedDemographics = false,
-  isEmergencyContactEnabled = false,
-}) => {
+const createForm = () => {
+  return PRE_CHECK_IN_FORM_PAGES.map(page => page.url);
+};
+
+const updateForm = (patientDemographicsStatus, isEmergencyContactEnabled) => {
   let pages = PRE_CHECK_IN_FORM_PAGES.map(page => page.url);
-  if (hasConfirmedDemographics) {
-    const skippedPages = [
-      URLS.DEMOGRAPHICS,
-      URLS.NEXT_OF_KIN,
-      URLS.EMERGENCY_CONTACT,
-    ];
-    pages = pages.filter(page => !skippedPages.includes(page));
-  }
   if (!isEmergencyContactEnabled) {
     pages = pages.filter(page => page !== URLS.EMERGENCY_CONTACT);
   }
+  const skippedPages = [];
+  const {
+    demographicsNeedsUpdate,
+    demographicsConfirmedAt,
+    nextOfKinNeedsUpdate,
+    nextOfKinConfirmedAt,
+    emergencyContactNeedsUpdate,
+    emergencyContactConfirmedAt,
+  } = patientDemographicsStatus;
+
+  const skipablePages = [
+    {
+      url: URLS.DEMOGRAPHICS,
+      confirmedAt: demographicsConfirmedAt,
+      needsUpdate: demographicsNeedsUpdate,
+    },
+    {
+      url: URLS.NEXT_OF_KIN,
+      confirmedAt: nextOfKinConfirmedAt,
+      needsUpdate: nextOfKinNeedsUpdate,
+    },
+    {
+      url: URLS.EMERGENCY_CONTACT,
+      confirmedAt: emergencyContactConfirmedAt,
+      needsUpdate: emergencyContactNeedsUpdate,
+    },
+  ];
+  const hoursTillExpire = 24;
+
+  const now = Date.now();
+
+  skipablePages.forEach(page => {
+    const pageLastUpdated = page.confirmedAt
+      ? new Date(page.confirmedAt)
+      : null;
+    if (
+      pageLastUpdated &&
+      Math.abs(now - pageLastUpdated) / 36e5 <= hoursTillExpire &&
```

Comment:
This line is doing the 24 hr check, my thoughts are: 

1. There might be a better way to do this with date-fns. 
2. Maybe this would be good to move to a function called like `isWithInHours(24, now, pageLastUpdated)`

Just kinda thinking out loud.

Reply:
ah both good points, forgot to look into the date lib we were already using. Some handy things in there.